### DISCUSSION TITLE
Address already in use: connect error after upgrading to MC 4.2

### DISCUSSION THREAD

**Discussion by rjkroll** [nextgenhealthcare/connect#5703] [Q&A] [answered]
Hi,

Weird issues that I am seeing for the first time today.    We have 10 channels, all of which monitor a folder and look for specifically named CSV files from CMS.    We parse the data, then on the destination, assign to variables, then execute a procedure on a SQL server with the CSV data as paramaters.

Normally, we used to be able to just start all 10 channels, they would import their CSV's fine, all to the same server database.

Today is the first time we tried these channels again and if I start them all at the same time, they will randomly produce hundreds of errors that say: Wrapped java.sql.SQLException: Network error IOException: Address already in use: connect

Thousand of records import without this error, but the only way for me to import all 10 files successfully as of right now is to start each channel 1 at a time, wait for it to finish, then start the next channel and repeat.  

Has something changed from Mirth 3.x to Mirth 4.2 that would cause this?  Is it somehow a SQL thing?   

I thought it might have to do with Max Threads, so I tried 1 and 10 and both produced the same results if I try to run more than one channel at a time.  

Additionally, we also moved from Java to OpenJDK, so I am not sure if that makes a difference or not.

```
CHANNEL:	CMS_CCLF7_Inbound_FS_v2_CKD
CONNECTOR:	Database Insert
SOURCE CODE:	
971: 	
972: 	var sqlStatement = "exec cclf.prc_insert_cclf7_v2 '"+$('CUR_CLM_UNIQ_ID')+"','"+$('BENE_HIC_NUM')+"','"+$('CLM_LINE_NDC_CD')+"','"+$('CLM_TYPE_CD')+"','"+$('CLM_LINE_FROM_DT')+"','"+$('PRVDR_SRVC_ID_QLFYR_CD')+"','"+$('CLM_SRVC_PRVDR_GNRC_ID_NUM')+"','"+$('CLM_DSPNSNG_STUS_CD')+"','"+$('CLM_DAW_PROD_SLCTN_CD')+"','"+$('CLM_LINE_SRVC_UNIT_QTY')+"','"+$('CLM_LINE_DAYS_SUPLY_QTY')+"','"+$('PRVDR_PRSBNG_ID_QLFYR_CD')+"','"+$('CLM_PRSBNG_PRVDR_GNRC_ID_NUM')+"','"+$('CLM_LINE_BENE_PMT_AMT')+"','"+$('CLM_ADJSTMT_TYPE_CD')+"','"+$('CLM_EFCTV_DT')+"','"+$('CLM_IDR_LD_DT')+"','"+$('CLM_LINE_RX_SRVC_RFRNC_NUM')+"','"+$('CLM_LINE_RX_FILL_NUM')+"','"+escoid+"','"+$('BENE_MBI_ID')+"','"+$('PHARM_TYPE')+"'"
973: 	//logger.error(sqlStatement)
974: 	var dbConn = DatabaseConnectionFactory.createDatabaseConnection('net.sourceforge.jtds.jdbc.Driver','jdbc:jtds:sqlserver://SQLTEST:5000/extract_db','user','pass');
975: 	var result = dbConn.executeCachedQuery(sqlStatement)
976: 	result.next()
977: 	if (result.getString("ERROR_NUMBER")!='1')
978: 	{
979: 		errorExists = 1
980: 		errorMessage = errorMessage + "\n\nERROR_NUMBER:  " + result.getString("ERROR_NUMBER") + "\nERROR_PROCEDURE:  " + result.getString("ERROR_PROCEDURE") + "\nERROR_MESSAGE: " + result.getString("ERROR_MESSAGE") + "\nSQL_STATEMENT: " + sqlStatement
LINE NUMBER:	976
DETAILS:	**Wrapped java.sql.SQLException: Network error IOException: Address already in use: connect**
```

**Comment by pacmano1** [comment]


**Reply by pacmano1** [reply]
Man, my bad... I apologize... I need a nap.  Let me delete my crap comments.

**Reply by rjkroll** [reply]
No worries! Its all good, I removed my comments to as to not clutter the thread

**Reply by pacmano1** [reply]
Do you have a lot of open or stale connections to MSSQL?   e.g. netstat from the command line.

**Reply by pacmano1** [reply]
or to anywhere? i.e. lots and lots of connections in assorted statuses.

**Reply by rjkroll** [reply]
Thanks, I just ran it and it literally took 30 seconds to finish outputting and I do see (i couldnt count them, so ill guess) hundreds of connections to that specific server, all that just say TIME_WAIT in the last column.    Currently, I am not even processing any files to that server, so it does seem wierd that they would all be there.    I am not sure how to get rid of them, though.   Service restart perhaps?

**Reply by pacmano1** [reply]
I am rusty in windows stuff but I think you can ```netsh``` stuff and/or modify the registry to allow more connections from your server running mirth

**Comment by pacmano1** [comment, marked as answer]
@rjkroll I think this was answered in the preceding comments in the thread, correct?

**Reply by rjkroll** [reply]
Yes it was, thanks.   I forgot to come back and mark it as answered.
